ConfigurationSystem/test/test_agentOptions.py

```diff
@@ -0,0 +1,83 @@
+          ('DIRAC.TransformationSystem.Agent.InputDataAgent', ['DateKey', 'TransformationTypes']),
+          ('DIRAC.TransformationSystem.Agent.MCExtensionAgent', ['TransformationTypes', 'TasksPerIteration',
+                                                                 'MaxFailureRate', 'MaxWaitingJobs']),
+          ('DIRAC.TransformationSystem.Agent.TaskManagerAgentBase', ['PluginLocation', 'BulkSubmission', 'shifterProxy',
+                                                                     'ShifterCredentials', 'maxNumberOfThreads']),
+          ('DIRAC.TransformationSystem.Agent.TransformationAgent', ['PluginLocation', 'transformationStatus',
+                                                                    'MaxFiles', 'MaxFilesToProcess',
+                                                                    'TransformationTypes', 'ReplicaCacheValidity',
+                                                                    'NoUnusedDelay', 'maxThreadsInPool']),
+          ('DIRAC.TransformationSystem.Agent.TransformationCleaningAgent', ['EnableFlag', 'shifterProxy']),
+          ('DIRAC.TransformationSystem.Agent.ValidateOutputDataAgent', ['TransformationTypes', 'DirectoryLocations',
+                                                                        'ActiveSEs', 'TransfIDMeta']),
+          # ('DIRAC.TransformationSystem.Agent.RequestTaskAgent', []),  # not inheriting from AgentModule
+          # ('DIRAC.TransformationSystem.Agent.WorkflowTaskAgent', []),  # not inheriting from AgentModule
+          ('DIRAC.WorkloadManagementSystem.Agent.JobAgent', ['FillingModeFlag', 'JobWrapperTemplate',
+                                                             'MinimumTimeLeft']),
+          ('DIRAC.WorkloadManagementSystem.Agent.JobCleaningAgent', []),
+          ('DIRAC.WorkloadManagementSystem.Agent.PilotStatusAgent', ['PilotAccountingEnabled', 'ClearPilotsDelay',
+                                                                     'ClearAbortedPilotsDelay']),
+          ('DIRAC.WorkloadManagementSystem.Agent.StalledJobAgent', ['StalledTimeHours', 'FailedTimeHours',
+                                                                    'StalledJobsTolerantSites', 'Enable']),
+          ('DIRAC.WorkloadManagementSystem.Agent.StatesAccountingAgent', []),
+          ('DIRAC.WorkloadManagementSystem.Agent.StatesMonitoringAgent', []),
+          ('DIRAC.WorkloadManagementSystem.Agent.SiteDirector', ['PilotLogLevel', 'SendPilotAccounting',
+                                                                 'MaxJobsInFillMode', 'FailedQueueCycleFactor', 'Site',
+                                                                 'GetPilotOutput', 'CETypes', 'ExtraPilotOptions',
+                                                                 'GridEnv', 'VO', 'Community', 'Group', 'Pilot3']),
+          # ('DIRAC.WorkloadManagementSystem.Agent.MultiProcessorSiteDirector', []),  #does not inherit from AgentModule
+          ]
```

Comment:
So, now every time that there is a new or modified option for an agent, such option should be added here?

Reply:
When a new option is added in the code, there will be an error if it is not also added in the ConfigTemplate.cfg
For the existing options, we need to fix the ConfigTemplates or the Agent code so that things are consistent.

Ideally all these lists would be empty, but in some cases it might be desirable to have the default option be different in the ConfigTemplate and the agent's code (DryRun for the VOMS2CSAgent for example).

Reply:
So, what do the lists of options here represent? Those options that we ignore to check (for now)?
I am just looking at one example here: SiteDirector. Shouldn't be the list shorter?

Reply:
Yes, right now it it all options that fail the test. Either because they don't exist in the ConfigTemplate, they are not called with am_getOption in `__init__` or initialize, or ebcause the value differs between the to.

For the SiteDirector, many of the options are not obtained  in `__init__` or `initialize` but in `beginExecution`. I could add a call to `beginExecution` to also get the calls to `am_getOption` there

Reply:
It probably makes sense, but then should you check also in `execute`?